clippy_lints/src/utils/higher.rs

```diff
@@ -133,11 +134,11 @@ pub fn is_from_for_desugar(local: &hir::Local<'_>) -> bool {
     false
 }
 
-/// Recover the essential nodes of a desugared for loop:
-/// `for pat in arg { body }` becomes `(pat, arg, body)`.
+/// Recover the entire span and the essential nodes of a desugared for loop:
+/// `for pat in arg { body }` becomes `(pat, arg, body)`. Return `(span, pat, arg, body)`.
 pub fn for_loop<'tcx>(
     expr: &'tcx hir::Expr<'tcx>,
-) -> Option<(&hir::Pat<'_>, &'tcx hir::Expr<'tcx>, &'tcx hir::Expr<'tcx>)> {
+) -> Option<(Span, &hir::Pat<'_>, &'tcx hir::Expr<'tcx>, &'tcx hir::Expr<'tcx>)> {
```

Comment:
A nit but I think the Span is more commonly tacked on at the end of the tuple. Thanks for updating this!